Assignments/ip1/ip1.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: assignment
+title: "Individual Project 1"
+permalink: /assignments/ip1
+parent: Assignments
+nav_order: 1
+due_date: "Wednesday September 18, 2024 12:00pm EST (Noon)"
+submission_notes: TBA
+---
+
+Welcome aboard to the Stack Overflow team! We're glad that you're here and ready to join our development team as a new software engineer.
+We're building an interactive application for online community to share their knowledge and experience, and are very happy to see that we have so many new developers who can help make this application a reality.
+By the end of the semester, you'll be able to propose, design, implement and test a new features for our project.
+We understand that some of you may have some web development experience, but don't expect that most of you do, and hence, have created an individual project to help you get up to speed with our existing codebase and development environment.
+
+FakeStackOverFlow is a web application that consists of some code that runs in each client's web browser, and also code that runs on a server.
+**TOUPDATE:** Users join the application in a "town": a 2D arcade-style map with different rooms to explore.
+Each town is also a video call: when two players get close to each other, they can see and hear each other; there is also a text chat available within the town.
+In Winter of 2021, our lead software engineer, Avery, developed a prototype for Covey.Town, and since then, hundreds of students in this class have built on that codebase.
+The most recent class-wide effort, in Fall 2023, added a concept called [Game Areas](https://neu-se.github.io/CS4530-Fall-2023/assignments/ip1), allowing players to play games within a special part of the town. 
+Students implemented a single game, [Tic-Tac-Toe](https://en.wikipedia.org/wiki/Tic-tac-toe), as a proof-of-concept for this feature.
+
+The objective for this semester's individual project is to **TOUPDATE**.
+This implementation effort will be split across two deliverables. In this first deliverable, you will implement and test the core backend components for this feature, and in the second deliverable, you will implement and test the frontend components. 
+
+## Objectives of this assignment
+The objectives of this assignment are to:
+*  Get you familiar with the basics of TypeScript, VSCode, and the project codebase
+*  Learn how to read and write code in TypeScript
+*  Translate high-level requirements into code
+*  Learn how to write unit tests with Jest
+
+## Getting started with this assignment
+
+Before you begin, be sure to check that you have NodeJS 18.x installed, along with VSCode. We have provided a tutorial on setting up a development environment for this class. 
```

Comment:
We should probably ask them to install the LTS verison

Reply:
We had some discussion on this point at our last meeting. Can you confirm that IP1 runs with the LTS version 20.16.1 ?